### PR TITLE
Added List<String> UriInfo#getMatchedResourceTemplates()

### PR DESCRIPTION
**Pull request by jansupol**

### CODE REVIEW

**Comment by jansupol** [comment]
Alternative to #1236 with a List return type

File: jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/jaxrs40/ee/rs/core/uriinfo/UriInfoTestResource.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+@Path("/resource")
+public class UriInfoTestResource {
+    public static final String ONE_POST = "{name:[a-zA-Z][a-zA-Z_0-9]*}";
+    public static final String TWO_GET = "{Prefix}{p:/?}{id: ((\\d+)?)}/abc{p2:/?}{number: (([A-Za-z0-9]*)?)}";
+    public static final String TWO_POST = "{Prefix}{p:/?}{id: ((\\d+)?)}/abc/{yeah}";
+    public static final String THREE_SUB = "{x:[a-z]}";
+
+    public static class SubGet {
+        @PUT
+        public String get(@Context UriInfo uriInfo) {
```
Comment (jim-krueger) [inline jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/jaxrs40/ee/rs/core/uriinfo/UriInfoTestResource.java:38]: Unless I'm mistaken, in order to fully test what Andrew requested this method would need a @Path(/something)

**Review by jim-krueger — DISMISSED** [review]

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/UriInfo.java [hunk 2]
```diff
@@ -246,6 +249,71 @@ public interface UriInfo {
...
+     * </tr>
+     * <tr>
+     * <td>GET /foo/bar/id0</td>
+     * <td>BarResource.getBar</td>
+     * <td>/foo, /{bar:[b-e][a-z]*}, /{id}{id:[0-9]}</td>
+     * </tr>
+     * </table>
+     *
+     * In case the method is invoked prior to the request matching (e.g. from a pre-matching filter), the method returns an
+     * empty string.
```
Comment (jamezp) [inline jaxrs-api/src/main/java/jakarta/ws/rs/core/UriInfo.java:311]: An empty list in this case.

**Review by spericas — APPROVED** [review]
LGTM

**Comment by jansupol** [comment]
@jamezp Do you prefer #1236 over this one?

**Comment by jamezp** [comment]
> @jamezp Do you prefer #1236 over this one?

I do prefer #1236 mainly because it seems to be what the OTel users are looking for. I'll admit it's easier for RESTEasy too, but we could make it work with a collection as well if we needed.

**Comment by jansupol** [comment]
To me, this one looks more universal, but perhaps it is unnecessary. We can go with the simpler one.

**Comment by jim-krueger** [comment]
FWIW:  This was the response from issue originator (Andrew) on this:  

> ... For our purposes, a String is good, if it was a list we'd want to concatenate it back into a single string.

**Comment by jansupol** [comment]
Closing this in favor of #1236

**Pull request closed without merging**